src/librustc_typeck/check/op.rs

```diff
@@ -562,7 +562,11 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                    on the left and may require reallocation. This \
                    requires ownership of the string on the left";
 
-        let is_std_string = |ty| &format!("{:?}", ty) == "std::string::String";
+        let string_type = self.tcx.get_diagnostic_item(sym::string_type);
+        let is_std_string = |ty: Ty<'tcx>| match &ty.ty_adt_def() {
```

Comment:
```suggestion
        let is_std_string = |ty: Ty<'tcx>| match ty.ty_adt_def() {
```

I don't think this ref is needed, is it?